pkg/apis/core/v1/storagecluster.go

```diff
@@ -527,6 +529,13 @@ type DataProviderSpec struct {
 	Params map[string]string `json:"params,omitempty"`
 }
 
+// GitOpsSpec contains the details for GitOps provider like github or bitbucket
+type GitOpsSpec struct {
+	Name   string                 `json:"name,omitempty"`
+	Type   string                 `json:"type"`
+	Params map[string]interface{} `json:"params"`
```

Comment:
nit: Let's add `omitempty`

Reply:
I skipped that intentionally. Autopilot can't instantiate a gitops provider w/o params provided

Reply:
The omit empty will just not show the params field if not specified. Dropping it won't help autopilot anyway. It is not going to force the user to specify params here.